src/main/java/com/dell/pravega/perf/performance.java

```diff
@@ -0,0 +1,18 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package com.dell.pravega.perf;
+
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+
+public interface performance {
+    void benchmark() throws InterruptedException, ExecutionException, IOException;
```

Comment:
It might be better to unwrap the execution exception and throw the underlying issue. 
Also IntruptedException is messy to deal with. (See `Exceptions.handleInterupted()` in pravega common.) I don't think a caller that did not request a blocking operation should necessarily have to handle that.
Normally interupted is only ever used to trigger an early shutdown. Maybe the benchmark should support that, but I'm not sure how that should affect this interface.

Reply:
yes , its done